pylint/graph.py

```diff
@@ -84,25 +85,36 @@ def generate(self, outputfile=None, mapfile=None):
 
         :rtype: str
         :return: a path to the generated file
+        :raises RuntimeError: if the executable for rendering was not found
         """
+        graphvis_extensions = ("dot", "gv")
         name = self.graphname
         if outputfile is not None:
             _, _, target = target_info_from_filename(outputfile)
-            if target != "dot":
-                pdot, dot_sourcepath = tempfile.mkstemp(".dot", name)
+            if not target:
+                target = "png"
+                outputfile = outputfile + "." + target
+            if target not in graphvis_extensions:
+                pdot, dot_sourcepath = tempfile.mkstemp(".gv", name)
                 os.close(pdot)
             else:
                 dot_sourcepath = outputfile
         else:
             target = "png"
-            pdot, dot_sourcepath = tempfile.mkstemp(".dot", name)
+            pdot, dot_sourcepath = tempfile.mkstemp(".gv", name)
             ppng, outputfile = tempfile.mkstemp(".png", name)
             os.close(pdot)
             os.close(ppng)
         pdot = codecs.open(dot_sourcepath, "w", encoding="utf8")
         pdot.write(self.source)
         pdot.close()
-        if target != "dot":
+        if target not in graphvis_extensions:
+            if shutil.which(self.renderer) is None:
+                raise RuntimeError(
+                    f"Cannot generate `{outputfile}` because '{self.renderer}' "
+                    "executable not found. Install Graphvis, or specify a `.gv` "
```

Comment:
Isn't it Graphviz ?

Reply:
Sorry, I don't understand

Reply:
I think that the lib name is ``graphviz`` not ``graphvis``.

Reply:
ohhhhhh right! I reread this so many times and never noticed the difference 😨